[css-pseudo-4] Clarify text decorations in ::spelling-error and ::grammar-error

schenney-chromium opened the pull request:
Issue #9233
Add a note for authors on the requirements for custom text-decorations inside ::spelling-error pseudo (and grammar).

fantasai reviewed (changes requested):
This is way too long for the problem it's solving.

fantasai commented:
I switched up the rules to use `text-decoration` instead of `text-decoration-line` to kind-of nudge people to use the shorthand also when trying to overwrite it. We could add a note, but this behavior should be very explicitly specified in the definition of these values; so I'm not sure how much we really need a note in this appendix here. But definitely we should keep it minimal if we add anything.

schenney-chromium commented:
I think something is needed because authors are quite likely to overlook this, just setting a `text-decoration-color` and expecting the UA rendered dots or squiggles to pick up the color, at least given the way Chrome renders error markers.

I simplified it to be more direct and moved it up to the section discussing properties in highlight pseudos.

This is basically a bandaid on the disconnect between `::spelling-error` defined here and `text-decoration-line: spelling-error` defined in the text-decor spec.
